### PR TITLE
RR-589 - Remove feature toggle PLP_PAGES_ENABLED and resultant redundant code

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
### Strap yourselves in! - this is a 102 file monster!
Strap yourselves in! - this is a 102 file monster, though I'm hoping the description and narrative below will help set the scene, and I'm also hoping you'll see the vast majority of change here is the deletion of entire files, or large swathes of code from files.

### Background
The CIAG dev team originally wrote the CIAG UI to include the Prisoner List page and the pre-induction Overview page, because that's what the original designs and scope of their development was to include.
We proposed the move of these 2 ages into the PLP UI, and as part of the development and migration for this we needed the CIAG dev team to introduce a feature toggle into their code base.
The feature toggle allows for switching between using their Prisoner List and pre-induction Overview pages, or ours.
The feature toggle has been enabled (ie. use the PLP pages) for some time now, and therefore the original CIAG Prisoner List page and the pre-induction Overview page (and associated nunjucks, service, client, utility etc code) have been sat unused, effectively in the `else` block of the feature toggle.

### Changes
This PR removes the feature toggle env var from the helm deployments and the `config.ts` file. Any code that was dependant on the feature toggle has been stripped out, to just leave the code to use the PLP UI pages (effectively just a few UI redirects). It was very much a case of following threads of code paths and removing as I went, but in summary:

* The nunjucks views for the Prisoner List page and the pre-induction Overview page have been removed entirely.
* Service code associated with producing the Prisoner List page and its associated client code (specifically prisoner-search API, CIAG API `/ciag/inductions/list` and PLP API `/action-plan/list` have all been removed
* Service code associated with producing the Overview page and it's associated client code (specifically Curious integrations) have all been removed
* Integration test code, specifically stubs and associated code for the above mentioned API calls, have all been removed
* Integration tests for the Prisoner List and Overview pages have been removed entirely
* Unused integration test stubs (other than those mentioned above) have been removed
* Unused environment variables (other than the feature toggle) have been removed from the helm deployments, `feature.env` and the readme

The only bit I've added are integration test stubs for the front end components library (every test was logging errors in the runtime console because the test could not get the header/footer; and a stub to represent the PLP UI

### CODE REVIEW

File: README.md
Comment (nathanrussell-moj-digital) [inline README.md:50]: This was simply wrong! 🤦

File: cypress.config.ts
Comment (nathanrussell-moj-digital) [inline cypress.config.ts:37]: `educationAndWorkPlanApi` stub is now unnecessary as this contained stubs for the goal and action plan based APIs.
The induction APIs (that were originally in the CIAG API) are still defined in `ciagApi` stubs. Clearly this is wrong as these APIs now live in PLP / Education & Work Plan; but perhaps we'll rename that file another day 👍 

`learningPlanUi` stub has been removed - it contained a simple/basic stub for any PLP page. I've moved it into a new file `plpUi` which contains stubs for specific PLP UI pages that the integration tests might end up redirecting the user to 👍

File: integration_tests/e2e/abilityToWork.cy.ts
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/abilityToWork.cy.ts:20]: These three stubs have been removed from pretty much every test. There were there to support the original CIAG UI Prisoner List page, in that it needed to call prisoner search to get a list of prisoners for the caseload ID, PLP's action plan list API, and CIAG API's induction list.

Because this UI no longer contains a prisoner list page, and therefore the tests never visit the page, these stubs are not required.

File: integration_tests/e2e/abilityToWork.cy.ts
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/abilityToWork.cy.ts:19]: These stubs are new stubs and have been added to pretty much every test.
They stub the PLP UI Prisoner List page; and the front end components API calls

File: integration_tests/e2e/login.cy.ts
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/login.cy.ts:27]: "Signing in" and rendering an page is no longer a responsibility of the CIAG UI, so many of these tests have been deleted.
After signing in to the CIAG UI, we now expect a redirect to the PLP UI. Given that its a redirect and the page itself (and its associated markup etc) is the responsibility of the PLP UI, we cannot (and there is no point) testing things like whether the username is displayed etc.
All of that is tested in the PLP UI 👍

File: integration_tests/e2e/updateFunctionalityFull.cy.ts
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/updateFunctionalityFull.cy.ts:31]: This was the original stub for "any PLP page". It has been replaced by specific stubs and `Page` class implementations, and the tests have been updated to use the correct stub and assert the correct page.

File: integration_tests/e2e/workPlan.cy.ts
Comment (nathanrussell-moj-digital) [inline integration_tests/e2e/workPlan.cy.ts:1]: "Work Plan" was the original CIAG UI pre-induction Overview page. 
Everything about that page has been removed, and so has this test 👍

File: server/config.ts
Comment (nathanrussell-moj-digital) [inline server/config.ts:101]: I've changed the relevant client classes to use the properties from the `educationAndWorkPlanApi` object, rather than the `ciagApi` object.
(They pointed to the same thing after we migrated the original CIAG API to the PLP API)

**Review by mattwills-moj-digital — APPROVED** [review]
I can't claim to have read all 102 affected files, but like you said, most of the changes involve deleting code and I think the rest of it looks good! 👍

**Comment by nathanrussell-moj-digital** [comment]
Thanks for the ✅ @mattwills-moj-digital 👍 

I've now marked this as Do Not Merge because I'm going to move the deployment of the CIAG UI to our `prod` namespace tonight, and I dont want to confuse that change (and any fall our from it ... which I hope is none! 🤞 ) with the change in this PR.

I'll look to merge this once the current CIAG UI codebase is deployed to our namespace and the dust has settled etc 👍

**Review by chrimesdev — APPROVED** [review]

**Comment by nathanrussell-moj-digital** [comment]
The move of the CIAG UI to the PLP namespace in `prod` has now happened and is working fine.
We are OK to merge this now 👍

**Pull request merged**
